mempool/reactor.go

```diff
@@ -49,6 +53,8 @@ func NewReactor(config *cfg.MempoolConfig, mempool *CListMempool, waitSync bool)
 		memR.waitSyncCh = make(chan struct{})
 	}
 	memR.mempool.SetTxRemovedCallback(func(txKey types.TxKey) { memR.removeSenders(txKey) })
+	memR.activeConnectionsSemaphore = semaphore.NewWeighted(int64(memR.config.ExperimentalMaxUsedOutboundPeers))
```

Comment:
Should we gate this with "if memR.config.ExperimentalMaxUsedOutboundPeers > 0", similarly to what you're doing below?

Reply:
We could do that, but the impact would be negligible here. This way the code seems cleaner.